project/src/components/ApartmentCard/ApartmentCard.tsx

```diff
@@ -0,0 +1,93 @@
+import { generatePath, Link } from 'react-router-dom';
+import { AppRoute } from '../../const';
+import { RoomOffer } from '../../types/room-offer';
+import { getRating } from '../../util';
+import { CardButton } from '../CardButton';
+
+type ApartmentCardProps = {
+  card: 'main' | 'favorite',
+  roomOffer: RoomOffer,
+  onMouseOver?: (id: number) => void,
+  onMouseLeave?: (id: number) => void,
+};
+
+const properties = {
+  main: {
+    className: 'cities',
+    width: 260,
+    height: 200,
+    placeCardActive: '',
+    AppRoute: AppRoute.Room,
+  },
+  favorite: {
+    className: 'favorites',
+    width: 150,
+    height: 110,
+    placeCardActive: 'place-card__bookmark-button--active',
+    AppRoute: AppRoute.Favorites,
+  },
+};
+
+function ApartmentCard({
+  roomOffer,
+  onMouseOver,
+  onMouseLeave,
+  card,
+}: ApartmentCardProps) {
+  const { id, previewImage, isPremium, price, rating, title, type } = roomOffer;
+  const { className, width, height, placeCardActive } = properties[card];
+
+  const ratingValue = getRating(rating);
+
+  return (
+    <article
+      className={`${className}__places-card place-card`}
+      onMouseOver={() => {
+        if (onMouseOver !== undefined) {
+          onMouseOver(id);
+        }
+      }}
```

Comment:
Более короткая запись 
```suggestion
      onMouseOver={() => onMouseOver && onMouseOver(id)}
```
Либо с использованием optional chaining
```suggestion
      onMouseOver={() => onMouseOver?.(id)}
```